lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -355,6 +361,13 @@ def cleanup():
                 logger.error("Service previously started but now misses the snap.")
                 return
 
+        # Store the current perf. profile we are applying
+        self.peers_data.put(
+            Scope.UNIT,
```

Comment:
It is hard to make sense of the current performance profile just by pooling the raw heap size and other sources. Better to just store it in the peer data.

With that said, may be worthy to save it on the app level.